### PR TITLE
Clean up to remove defers from loops

### PR DESCRIPTION
**Pull request by ravisantoshgudimetla**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind cleanup

**What this PR does / why we need it**:
A small clean up to remove defers from loops and function renaming.

/cc @Huang-Wei @bsalamat 

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: test/e2e/scheduling/priorities.go [hunk 1]
```diff
@@ -219,16 +219,27 @@ var _ = SIGDescribe("SchedulerPriorities [Serial]", func() {
 		//we need apply more taints on a node, because one match toleration only count 1
 		ginkgo.By("Trying to apply 10 taint on the nodes except first one.")
 		nodeName := nodeList.Items[0].Name
-
+		var totalTaints []*v1.Taint
```
Comment (bsalamat) [inline test/e2e/scheduling/priorities.go]: nit: s/totalTaints/addedTaints/

File: test/e2e/scheduling/priorities.go [hunk 2]
```diff
@@ -219,16 +219,27 @@ var _ = SIGDescribe("SchedulerPriorities [Serial]", func() {
...
 			}
 			for i := 0; i < 10; i++ {
-				testTaint := addRandomTaitToNode(cs, node.Name)
-				defer framework.RemoveTaintOffNode(cs, node.Name, *testTaint)
+				testTaint := addRandomTaintToNode(cs, node.Name)
+				totalTaints = append(totalTaints, testTaint)
 			}
 		}
+		defer func() {
+			for index, node := range nodeList.Items {
```
Comment (bsalamat) [inline test/e2e/scheduling/priorities.go]: A more robust approach that would tolerate addition/removal of nodes is to iterate over `totalTaints` and remove them. In order to do so, you will need to change the slice to a map:

`addedTaints map[string][]*v1.Taint`

Key of the map will be a node name.

Comment (ravisantoshgudimetla) [reply test/e2e/scheduling/priorities.go]: Ok, I will address them, thanks @bsalamat

File: test/e2e/scheduling/priorities.go [hunk 3]
```diff
@@ -219,16 +219,27 @@ var _ = SIGDescribe("SchedulerPriorities [Serial]", func() {
 		//we need apply more taints on a node, because one match toleration only count 1
```
Comment (Huang-Wei) [inline test/e2e/scheduling/priorities.go]: Can you help add a space after `//`?

File: test/e2e/scheduling/priorities.go [hunk 4]
```diff
@@ -243,11 +254,18 @@ var _ = SIGDescribe("SchedulerPriorities [Serial]", func() {
 
 		ginkgo.By("Trying to apply 10 taint on the first node.")
 		var tolerations []v1.Toleration
+		var alltaintsForSingleNode []*v1.Taint
 		for i := 0; i < 10; i++ {
-			testTaint := addRandomTaitToNode(cs, nodeName)
+			testTaint := addRandomTaintToNode(cs, nodeName)
 			tolerations = append(tolerations, v1.Toleration{Key: testTaint.Key, Value: testTaint.Value, Effect: testTaint.Effect})
-			defer framework.RemoveTaintOffNode(cs, nodeName, *testTaint)
+			alltaintsForSingleNode = append(alltaintsForSingleNode, testTaint)
```
Comment (Huang-Wei) [inline test/e2e/scheduling/priorities.go]: s/alltaintsForSingleNode/allTaintsForFirstNode/

**Pull request closed without merging**